stdlib/test/builtin/test_string.mojo

```diff
@@ -1107,6 +1098,93 @@ def test_indexing():
     assert_equal(a[2], "c")
 
 
+def test_string_iter():
+    var vs = String("123")
+
+    # Borrow immutably
+    fn conc(vs: String) -> String:
+        var c = String("")
+        for v in vs:
+            c += v
+        return c
+
+    assert_equal(123, atol(conc(vs)))
+
+    concat = String("")
+    for v in vs.__reversed__():
+        concat += v
+    assert_equal(321, atol(concat))
+
+    # TODO: UnsafePointer does not have a store or __setitem__ method
+    # for v in vs:
+    #     v.unsafe_ptr().store(0, "1")
```

Comment:
**Suggestion** I think this could be written as `v.unsafe_ptr()[0] = "1"`.

`UnsafePointer` does define a `__getitem__` method, [seen here](https://github.com/modularml/mojo/blob/388cbdb8de2e174175f1b0ff8e88aa85da2d68c7/stdlib/src/memory/unsafe_pointer.mojo#L197-L199).

(However, that method was previously called `__refitem__`, and was only changed to `__getitem__` using new language features in the last few days; so if you searched for `__getitem__` that's why you may not have seen it 🙂.)